e2e/autoscript-apps/src/authn-central-login-no-iframe/autoscript.ts

```diff
@@ -0,0 +1,145 @@
+
+  const url = new URL(window.location.href);
+  const amUrl = url.searchParams.get('amUrl') || 'http://localhost:9443/am';
+  const preAuthenticated = url.searchParams.get('preAuthenticated') || 'false';
+  const code = url.searchParams.get('code') || '';
+  const clientId = url.searchParams.get('clientId');
+  const client_id = url.searchParams.get('client_id');
+  const error = url.searchParams.get('error_description') || false;
+  const realmPath = url.searchParams.get('realmPath') || 'root';
+  const scope = url.searchParams.get('scope') || 'openid profile me.read';
+  const state = url.searchParams.get('state') || '';
+  const acr_values = url.searchParams.get('acr') || 'skipBackgroundRequest';
+  // in central login we use an auth query param for the return of our mock 401 request
+  // this is to prevent the evaluation of the page before we have technically authenticated
+  const auth = url.searchParams.get('auth') || false;
+
+  let tokenStore = url.searchParams.get('tokenStore') || 'localStorage';
+
+  // Support full redirects by setting storage, rather than rely purely on URL
+  if (!localStorage.getItem('tokenStore')) {
+    localStorage.setItem('tokenStore', tokenStore);
+  } else {
+    tokenStore = localStorage.getItem('tokenStore');
+  }
+
+  console.log('Configure the SDK');
+  forgerock.Config.set({
+    clientId: clientId || client_id || 'CentralLoginOAuthClient',
+    realmPath,
+    redirectUri: `${url.origin}/src/${
+      preAuthenticated === 'false' ? 'authn-central-login' : '_callback'
+    }/`,
+    scope,
+    serverConfig: {
+      baseUrl: amUrl,
+    },
+    tokenStore,
+  });
+
+  if (!code && !state) {
+    try {
+      forgerock.SessionManager.logout();
+    } catch (err) {
+      // Do nothing
+    }
+  }
+
+  console.log('Initiate first step with `undefined`');
+
+  // Wrapping in setTimeout to give the test time to bind listener to console.log
+  setTimeout(() => {
+    from([1])
+      .pipe(
+        map(() => {
+          if (preAuthenticated === 'true') {
+            console.log('Set mock cookie to represent existing session');
+            document.cookie = 'iPlanetDirectoryPro=abcd1234; domain=localhost; path=/';
+            if (code && state) {
+              window.sessionStorage.setItem(
+                `FR-SDK-${clientId}`,
+                JSON.stringify({ responseType: 'code', state, verifier: '1234' }),
+              );
+            }
+          }
+          return;
+        }),
+        rxDelay(delay),
+        mergeMap((step) => {
+          let tokens;
+          // detect when in iframe, throw as error if so
+          if (window.self !== window.top) {
+            throw new Error('Loaded_In_Iframe');
+          } else if (code && state) {
+            tokens = forgerock.TokenManager.getTokens({
+              query: { code, state, acr_values },
+            });
+          } else {
+            tokens = forgerock.TokenManager.getTokens({
+              skipBackgroundRequest: true,
```

Comment:
i'm concerned with this naming, i just fear another `modern` | `legacy` situation where people will read this name and say yeah we want to skip this.

Reply:
As mentioned on Slack. I think this is okay, and unintentially using this option won't do any harm. I've added more information in the JSDoc to cover this option's intention.